log/namespaced/namespaces_test.go

```diff
@@ -0,0 +1,86 @@
+package namespaced
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/assertions"
+)
+
+func TestNamespaces(t *testing.T) {
+	a := New(t)
+
+	ns := &ns{}
+
+	// empty namespace is accepted
+	{
+		ns.Set([]string{})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeFalse)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// add some namespaces
+	{
+		ns.Set([]string{
+			"a",
+			"b",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeFalse)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// * accepts everything
+	{
+		ns.Set([]string{
+			"*",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeTrue)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// negation wins from *
+	{
+		ns.Set([]string{
+			"*",
+			"-a",
+		})
+
+		a.So(ns.IsEnabled("a"), ShouldBeFalse)
+		a.So(ns.IsEnabled("b"), ShouldBeTrue)
+		a.So(ns.IsEnabled("c"), ShouldBeTrue)
+		a.So(ns.IsEnabled(""), ShouldBeTrue)
+	}
+
+	// order should not matter
+	{
+		ns.Set([]string{
+			"-a",
+			"-b",
+			"*",
+		})
+
```

Comment:
I find it weird (maybe it's me), but order should matter.
Deactivating `a` and `b` and activate everything should activate everything
Activating everything and deactivate `a` and `b` should indeed allow every component to log apart from `a` and `b`

I belive that order matters in this case.